app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -23,22 +24,23 @@
 import com.zerobranch.layout.SwipeLayout;
 
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
 
 public class ShowAdapter extends RecyclerView.Adapter<ShowAdapter.ViewHolder> {
     // instance variables
     private Context context;
     private List<Show> currentShows; // currently displayed shows on recycler view
-    private List<Show> allShows;     // list of all trending or recommended shows retrieved from Trakt
+    private List<Show> allShows; // all shows from Trakt call
+    private ShowFilter showFilter;
     private NavigationInterface nav;
 
-    public ShowAdapter (Context context, List<Show> shows, NavigationInterface nav){
+    public ShowAdapter (Context context, List<Show> shows, NavigationInterface nav, ShowFilter showFilter){
```

Comment:
So, I would have the show filter live outside of this. 

In the feed fragment and recommendation fragments, just have it filter and the shows and then tell the adapter to show the new filtered list